models/DataObject/Fieldcollection/Definition.php

```diff
@@ -246,7 +246,9 @@ public function isWritable(): bool
      */
     public function getDefinitionFile(string $key = null): string
     {
-        return $this->locateDefinitionFile($key ?? $this->getKey(), 'fieldcollections/%s.php');
+        $lowerKey = $key ?? $this->getkey();
+
+        return $this->locateDefinitionFile(strtolower($lowerKey), 'fieldcollections/%s.php');
```

Comment:
Seems strange to call `strtolower()` on a variable named `$lowerKey`.

Maybe write it like this instead?
```suggestion
        $key ??= $this->getkey();

        return $this->locateDefinitionFile(strtolower($key), 'fieldcollections/%s.php');
```